.github/workflows/build-docs.yml

```diff
@@ -23,10 +28,40 @@ jobs:
             ${{ runner.os }}-build-${{ env.cache-name }}-
             ${{ runner.os }}-build-
             ${{ runner.os }}-
-      - name: Build docs
+      - name: Cache benchmark node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules-bm
+        with:
+          path: amagaki-benchmark/node_modules
+          key: ${{ runner.os }}-build-bm-${{ env.cache-name }}-${{ hashFiles('amagaki-benchmark/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-bm-${{ env.cache-name }}-
+      - name: Build docs and benchmark build
         run: |
                 npm install --dev
+                npm run dev:start
+
+                # Build the docs into docs/
                 npm run docs
+
+                # Preserve benchmark files.
```

Comment:
Separate this out into a different step too, to make it like degu, and then I think we're good to go:  
https://github.com/blinkkcode/degu/blob/main/.github/workflows/docs.yml#L26